#**Quantitative Analysis of Stock Market**

The objective of this project is to gain deeper understanding of stock market dynamics by performing quantitative analysis to inform investment strategies.



**Goals**



*   Trend Analysis
*   Volatility Assessment
*   Correlation Study
*   Risk-Return Trade-off Analysis




In [1]:
from google.colab import drive

# Mount Google Drive
drive.mount('/content/drive')

Mounted at /content/drive


In [2]:
%cd /content/drive/MyDrive/Stock Market Analysis

/content/drive/MyDrive/Stock Market Analysis


In [10]:
import pandas as pd
import plotly.express as px
import plotly.graph_objects as go
from plotly.subplots import make_subplots
import plotly.io as pio
pio.templates.default = "plotly_white"

In [5]:
# Import the data
df = pd.read_csv('stocks.csv')
print(df.shape)
df.head()

(248, 8)


Ticker        Date        Open        High         Low       Close  \
0   AAPL  2023-02-07  150.639999  155.229996  150.639999  154.649994   
1   AAPL  2023-02-08  153.880005  154.580002  151.169998  151.919998   
2   AAPL  2023-02-09  153.779999  154.330002  150.419998  150.869995   
3   AAPL  2023-02-10  149.460007  151.339996  149.220001  151.009995   
4   AAPL  2023-02-13  150.949997  154.259995  150.919998  153.850006   

    Adj Close    Volume  
0  154.414230  83322600  
1  151.688400  64120100  
2  150.639999  56007100  
3  151.009995  57450700  
4  153.850006  62199000

The dataset contains the following features for stock market data:



*   Ticker: The stock ticker symbol.
*   Date: The specific trading date.
*   Open: Opening price of the stock for the day.
*   High: Highest price point of the stock during the day.
*   Low: Lowest price point during the day.
*   Close: Closing price of the stock.
*   Adj Close: Adjusted closing price, factoring in corporate actions like splits.
*   Volume: Total trading volume of the stock.

##**Descriptive Statistics**

To get the summary statistics for each stokc in the dataset.

In [7]:
df.groupby('Ticker')['Close'].describe()

count        mean        std         min         25%         50%  \
Ticker                                                                     
AAPL     62.0  158.240645   7.360485  145.309998  152.077499  158.055000   
GOOG     62.0  100.631532   6.279464   89.349998   94.702501  102.759998   
MSFT     62.0  275.039839  17.676231  246.270004  258.742500  275.810013   
NFLX     62.0  327.614677  18.554419  292.760010  315.672493  325.600006   

               75%         max  
Ticker                          
AAPL    165.162506  173.570007  
GOOG    105.962503  109.459999  
MSFT    287.217506  310.649994  
NFLX    338.899994  366.829987

Observations:

*   AAPL: Apple
The mean closing price is 158.24 and the standard deviation is 7.36.

*   GOOG: Google
The mean closing price is 100.63, with a standard deviation of 6.28, indicating less variability in clossing prices compared to AAPL.

*   MSFT: MicroSoft
It has higher mean closing price of 275.04 and a higher standard deviation of 17.28, suggesting greater price variability than AAPL and GOOG.

*   NFLX: Netflix
NFLX shows the highest mean closing price among these stock and the highest standard deviation, indicating the most significant price fluctuation.

##**Time Series Analysis**

Performing time series analysis to examine the trends and patterns over time

In [15]:
df['Date'] = pd.to_datetime(df['Date'])
pivot_df = df.pivot(index ='Date', columns = 'Ticker', values = 'Close')
pivot_df.head()

Ticker            AAPL        GOOG        MSFT        NFLX
Date                                                      
2023-02-07  154.649994  108.040001  267.559998  362.950012
2023-02-08  151.919998  100.000000  266.730011  366.829987
2023-02-09  150.869995   95.459999  263.619995  362.500000
2023-02-10  151.009995   94.860001  263.100006  347.359985
2023-02-13  153.850006   95.000000  271.320007  358.570007

In [12]:
fig = make_subplots(rows=1, cols=1)

for column in pivot_df.columns:
    fig.add_trace(
        go.Scatter(x=pivot_df.index, y=pivot_df[column], name=column),
        row=1, col=1
    )

fig.update_layout(
    title_text='Time Series of Closing Prices',
    xaxis_title='Date',
    yaxis_title='Closing Price',
    legend_title='Ticker',
    showlegend=True
)

fig.show()

Observations:

*   Each stock shows its unique trends over time. APPL and MSFT exhibit a general upward trend in this period.

*   There is noticeable volatility in the stock prices. NFLX shows more pronounced fluctuations compared to othes.

*   When comparing the stocks, MSFT and NFLX generally trade at higher price levels than AAPL and GOOG.

##**Volatility Analsis**

Performing volatility analysis to gain insights into how much the stock prices fluctuated over period.

In [16]:
volatility = pivot_df.std().sort_values(ascending=False)

fig = px.bar(volatility,
             x=volatility.index,
             y=volatility.values,
             labels={'y': 'Standard Deviation', 'x': 'Ticker'},
             title='Volatility of Closing Prices')
fig.show()

Observations:

NFLX has the highest volatility and GOOG is the least volatile in the dataset.

It indicates that NFLX and MSFT stocks were prone to price fluctuations during this period compared to AAPL and GOOG.

##**Correlation Analysis**

In [17]:
correlation_matrix = pivot_df.corr()

fig = go.Figure(data=go.Heatmap(
                    z=correlation_matrix,
                    x=correlation_matrix.columns,
                    y=correlation_matrix.columns,
                    colorscale='blues',
                    colorbar=dict(title='Correlation'),
                    ))

fig.update_layout(
    title='Correlation Matrix of Closing Prices',
    xaxis_title='Ticker',
    yaxis_title='Ticker'
)

fig.show()

Observations:

There are varying degrees of positive correlations between the stock prices, with some showing stronger correlations than others.

AAPL and MSFT seem to have relatively higher positive correlation.

##**Comparative Analysis**

We will compare the performance of different stocks based on their returns over the period.

In [18]:
# Calculating the percentage change in closing prices
percentage_change = ((pivot_df.iloc[-1] - pivot_df.iloc[0]) / pivot_df.iloc[0]) * 100

fig = px.bar(percentage_change,
             x=percentage_change.index,
             y=percentage_change.values,
             labels={'y': 'Percentage Change (%)', 'x': 'Ticker'},
             title='Percentage Change in Closing Prices')
fig.show()

Observations:

*   MSFT shows the highest positive change with 16.1%
*   AAPL exhibited a positive change of 12.23% though slightly lower than MSFT.
*   GOOG showed a slight negative change indicating a minor decline in its stock price over the observed period
*   NFLX experience the most significant negative change, suggesting a notable decrease in its stock price during the period.

##**Daily Risk Vs. Return Analysis**

In [19]:
daily_returns = pivot_df.pct_change().dropna()

# Recalculating average daily return and standard deviation (risk)
avg_daily_return = daily_returns.mean()
risk = daily_returns.std()

# Creating a DataFrame for plotting
risk_return_df = pd.DataFrame({'Risk': risk, 'Average Daily Return': avg_daily_return})

fig = go.Figure()

fig.add_trace(go.Scatter(
    x=risk_return_df['Risk'],
    y=risk_return_df['Average Daily Return'],
    mode='markers+text',
    text=risk_return_df.index,
    textposition="top center",
    marker=dict(size=10)
))

fig.update_layout(
    title='Risk vs. Return Analysis',
    xaxis_title='Risk (Standard Deviation)',
    yaxis_title='Average Daily Return',
    showlegend=False
)

fig.show()

Observations:



*   AAPL shows the lowest risk combined with a positive average daily return, suggesting a more stable investment with consistent returns.
*   GOOG has higher volatility than AAPL and, on average a slightly negative daily return, indicating a riskier and less rewarding investment during this period.
*   MSFT shows moderate risk with the highest daily return, suggesting a potentially more rewarding investment, although with higher volatility compared to AAPL.
*   NFLX exhibits the highest risk and a negative average daily return, indicating it was the most volatile and least rewarding investment among these stocks over the analyzed period.





##**Conclusion**

* Tech Giants Lead, But with Different Risk Profiles: While tech giants like Apple (AAPL), Microsoft (MSFT), Google (GOOG), and Netflix (NFLX) are prominent players in the market, they exhibit diverse risk profiles. While AAPL and MSFT offer stability and long-term growth potential, GOOG presents moderate risk with the potential for growth. NFLX, on the other hand, stands out for its high volatility, indicating a riskier investment.

* Trend Analysis Highlights Growth Potential: The upward trends observed in AAPL and MSFT suggest strong growth potential over the analyzed period. Investors seeking stable long-term investments may find opportunities in these stocks.

* Volatility Requires Strategic Management: High volatility, particularly evident in NFLX, underscores the importance of strategic risk management. Investors must employ tactics such as diversification or options strategies to mitigate the impact of price fluctuations.

* Correlation and Diversification: Positive correlations between AAPL and MSFT suggest that their prices tend to move in the same direction, indicating a need for diversified portfolios to manage risk effectively. Pairing assets with lower correlation can enhance portfolio stability and reduce overall risk.

* Balancing Risk and Return: MSFT stands out for its higher positive change and daily return, offering potentially higher rewards but with moderate risk. Conversely, AAPL presents a more conservative option with consistent positive returns and lower risk. Investors must weigh their risk tolerance against return expectations when crafting their investment strategies.

* Adapting to Market Dynamics: Market conditions and company performance evolve over time. Regular monitoring and adjustment of investment strategies are crucial to capitalize on emerging trends and mitigate risks effectively.

* Value of Informed Decision-Making: In-depth quantitative analysis provides valuable insights into stock market dynamics, enabling investors to make informed decisions aligned with their investment goals and risk preferences.